src/nvim/lua/executor.c

```diff
@@ -1909,3 +1910,186 @@ void nlua_do_ucmd(ucmd_T *cmd, exarg_T *eap)
     nlua_error(lstate, _("Error executing Lua callback: %.*s"));
   }
 }
+
+void ex_lsp(exarg_T *eap)
+{
+  lua_State *const lstate = global_lstate;
+
+  lua_getglobal(lstate, "require");
+  if (lua_type(lstate, -1) != LUA_TFUNCTION) {
+    lua_pop(lstate, 1);
+    emsg(_("require is not a function"));
+    return;
+  }
+
+  lua_pushstring(lstate, "vim.lsp.command");
+  if (nlua_pcall(lstate, 1, 1) != 0) {
+    nlua_error(lstate, _("Error executing vim.lsp.command: %.*s"));
+    return;
+  }
+  if (lua_type(lstate, -1) != LUA_TTABLE) {
+    lua_pop(lstate, 1);
+    emsg(_("vim.lsp.command is not a table"));
+    return;
+  }
+
+  lua_getfield(lstate, -1, "run");
+  if (lua_type(lstate, -1) != LUA_TFUNCTION) {
+    lua_pop(lstate, 1);
+    emsg(_("vim.lsp.command.run is not a function"));
+    return;
+  }
```

Comment:
we could load the string `"require('vim.lsp.command').run(...)"` instead and have luajit generate all errors for us, instead of the hand-coded prelude.